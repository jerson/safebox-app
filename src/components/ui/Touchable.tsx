import React from 'react';
import { TouchableOpacity, TouchableOpacityProps, View } from 'react-native';

export interface TouchableProps extends TouchableOpacityProps {
  forwardedRef?: any;
  children?: React.ReactNode[] | React.ReactChild;
}

function Touchable(props: TouchableProps, ref: React.Ref<TouchableOpacity>) {
  const { onPress, onLongPress, ...extraProps } = props;

  const hasAction =
    typeof onPress === 'function' || typeof onLongPress === 'function';
  if (!hasAction) {
    return <View ref={ref} {...extraProps} />;
  }
  return (
    <TouchableOpacity
      ref={ref}
      {...extraProps}
      onPress={onPress}
      onLongPress={onLongPress}
    />
  );
}

export default React.forwardRef(Touchable);
