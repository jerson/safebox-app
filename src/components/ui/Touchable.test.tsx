import React from 'react';
import * as renderer from 'react-test-renderer';
import Touchable, {TouchableProps} from './Touchable';

const props: TouchableProps = {
  onPress: jest.fn(),
  onLongPress: jest.fn(),
};

test('renders without crashing', () => {
  const snapshot = renderer.create(<Touchable {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<Touchable {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
