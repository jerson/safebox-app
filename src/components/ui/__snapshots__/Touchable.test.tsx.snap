// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="touchable"
/>
`;
